gobblin-cluster/src/main/java/org/apache/gobblin/cluster/HelixUtils.java

```diff
@@ -89,34 +90,43 @@ public static String getHelixInstanceName(
     return namePrefix + "_" + instanceId;
   }
 
+  // We have switched from Helix JobQueue to WorkFlow based job execution.
+  @Deprecated
   public static void submitJobToQueue(
       JobConfig.Builder jobConfigBuilder,
       String queueName,
       String jobName,
       TaskDriver helixTaskDriver,
       HelixManager helixManager,
       long jobQueueDeleteTimeoutSeconds) throws Exception {
+    submitJobToWorkFlow(jobConfigBuilder, queueName, jobName, helixTaskDriver, helixManager, jobQueueDeleteTimeoutSeconds);
+  }
 
+  public static void submitJobToWorkFlow(JobConfig.Builder jobConfigBuilder,
+      String queueName,
+      String jobName,
+      TaskDriver helixTaskDriver,
+      HelixManager helixManager,
+      long jobQueueDeleteTimeoutSeconds) throws Exception {
     WorkflowConfig workflowConfig = helixTaskDriver.getWorkflowConfig(helixManager, queueName);
-
     // If the queue is present, but in delete state then wait for cleanup before recreating the queue
     if (workflowConfig != null && workflowConfig.getTargetState() == TargetState.DELETE) {
       new TaskDriver(helixManager).deleteAndWaitForCompletion(queueName, jobQueueDeleteTimeoutSeconds);
       // if we get here then the workflow was successfully deleted
       workflowConfig = null;
     }
 
-    // Create one queue for each job with the job name being the queue name
+    // Create a work flow for each job with the name being the queue name
     if (workflowConfig == null) {
-      JobQueue jobQueue = new JobQueue.Builder(queueName).build();
-      helixTaskDriver.createQueue(jobQueue);
-      log.info("Created job queue {}", queueName);
+      // Create a workflow, set expiry to 5 minutes and add the job
```

Comment:
Can you put some documentation for this method to explain :
1) Why we want to deleteAndWaitForCompletion?
2) How the expiration works from Helix point of view?